tasks/Module 3/task_2.py

```diff
@@ -0,0 +1,57 @@
+"""Module2/task2: custom, python split function."""
+import re
+import pytest
+
+# Vars
+EMPTY_STRING_ERROR = 'Please provide a valid separator. ' \
+                     'It cannot be an empty string.'
+
+
+# Test data
+TEST_DATA_SET_1 = [
+    ('Python is cool', ' ', ['Python', 'is', 'cool']),
+    (',,,,', ',', ['', '', '', '', '']),
+    ('Python', 'Javascript', ['Python'])
+]
+TEST_DATA_SET_2 = [
+    ('feuer DPS pew pew', '')
+]
+
+
+def my_split(string_to_split, sep=None) -> list:
+    """Python's implementation of builtin (C) split function.
+
+    :param string_to_split: string that shall be splitted.
+    :param sep: separator by which string is splitted. None equals ' '
```

Comment:
> None equals ' '

It is not exactly how you function behaves. If sep is None you function does not split by single spaces, but by any whitespace character. Also, the built-in `str.split()` filters out empty strings if separator is not provided:
```python
>>> "double  space".split()  # Empty string removed
['double', 'space']
>>> "double  space".split(' ')  # Empty string preserved
['double', '', 'space']
```